src/rlookup.c

```diff
@@ -702,10 +707,14 @@ int RLookup_LoadDocument(RLookup *it, RLookupRow *dst, RLookupLoadOptions *optio
   } else {
     rv = loadIndividualKeys(it, dst, options);
   }
+  if (bc) {
+  // Unlock the GIL
+    RedisModule_ThreadSafeContextUnlock(ctx);
+  }
   // if loading the document failed b/c it does not exist, delete the document from DocTable
   // this will mark doc as deleted and reply with `(nil)`
   if (rv != REDISMODULE_OK) {
-    RedisModuleCtx *ctx = options->sctx->redisCtx;
+    //TODO: make sure we are not taking the write lock here while we are in read lock 
```

Comment:
Doesn't deleting a doc from the spec requires a write lock?

Reply:
yes, I moved the release down after the error check.
Note that this PR main purpose was to enable readers in parallel so the GIL locking here is very coarse. 
The flow will be different when writers are also enabled.